### PR TITLE
Enable more Transaction v5 tests

### PR DESCRIPTION
**Pull request by oxarbitrage**
## Motivation

Currently, most of Zebra's proptests are testing Transaction v4, but we also want to test Transaction v5 (and NU5).

## Solution

- make NU5 the default network upgrade for proptests
- use arbitrary `LedgerState`s in more tests
- stop skipping Canopy (and later) blocks in some tests

The code in this pull request has:
  - [x] Documentation Comments
  - [x] Property Tests

This is a replacement of https://github.com/ZcashFoundation/zebra/pull/2061

The automatic rebase didn't worked as expected, there are conflicts so i cherry picked the 2 commits involved and made a new PR.

## Review

@oxarbitrage can review. These tests don't pass without his fixes to #2023.

These fixes also depend on #2060.

## Follow Up Work

#2059 Use arbitrary LedgerStates in more tests to increase test coverage

### CODE REVIEW

**Review by oxarbitrage — COMMENTED** [review]
Looks good but i cant approve as i created the PR even if the commits are from @teor2345

**Comment by teor2345** [comment]
> Looks good but i cant approve as i created the PR even if the commits are from @teor2345

You can use admin-merge for cases like this.

**Review by teor2345 — APPROVED** [review]
This is my code and @oxarbitrage has reviewed it.

**Pull request merged**
